common/predictive-text/unit_tests/headless/worker-custom-punctuation.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Unit tests for the Dummy prediction model.
+ */
+
+var assert = require('chai').assert;
+var DummyModel = require('../../build/intermediate').models.DummyModel;
+var ModelCompositor = require('../../build/intermediate').ModelCompositor;
+
+describe('Custom Punctuation', function () {
+  it('appears in the keep suggestion', function () {
+    let dummySuggestions = [
+      {
+        transform: {
+          insert: 'Hrllo',
+          deleteLeft: 0,
+        },
+        tag: 'keep',
+        displayAs: 'Hrllo',
+      },
+      {
+        transform: {
+          insert: 'Hello',
+          deleteLeft: 0,
+        },
+        displayAs: 'Hello',
+      },
+      {
+        transform: {
+          insert: 'Jello',
+          deleteLeft: 0,
+        },
+        displayAs: 'Jello',
+      }
+    ];
+
+    var model = new DummyModel({
+      futureSuggestions: [dummySuggestions],
+      punctuation: {
+        quotesForKeepSuggestion: {
+          open: "«", close: "»"
+        },
+        insertAfterWord: " "
```

Comment:
I see.  While we can handle this automatically for spacebar-acceptances (whenever we enable those), this is needed to properly handle acceptance via banner selection.  Makes sense to model it here.